src/raiden_libs/service_registry.py

```diff
@@ -360,40 +442,106 @@ def maybe_prompt(query: str) -> None:
     log.info("Updated infos", current_url=current_url)
 
 
-@blockchain_options(contracts=[CONTRACT_DEPOSIT, CONTRACT_SERVICE_REGISTRY])
-@cli.command("withdraw")
-@click.option(
-    "--to",
-    type=str,
-    callback=validate_address,
-    help="Target address for withdrawn tokens",
-)
-@common_options("service_registry")
+def find_deposits(
+    web3: Web3,
+    service_address: Address,
+    service_registry_contract: Contract,
+    start_block: BlockNumber,
+) -> List[Dict[str, Any]]:
+    """
+    Return the address of the oldest deposit contract which is not withdrawn
+    """
+    # Get RegisteredService events for service_address
+    event_abi = dict(service_registry_contract.events[EVENT_REGISTERED_SERVICE]().abi)
+    topics = [
+        event_abi_to_log_topic(event_abi),
+        bytes([0] * 12) + service_address,
+    ]
+    filter_params = FilterParams(
+        {
+            "fromBlock": start_block,
+            "toBlock": "latest",
+            "address": service_registry_contract.address,
+            "topics": [HexStr("0x" + t.hex()) for t in topics],
+        }
+    )
+    raw_events = web3.eth.getLogs(filter_params)
+    events = [decode_event(service_registry_contract.abi, event) for event in raw_events]
+
+    # Bring events into a pleasant form
+    return [
+        dict(
+            block_number=e["blockNumber"],
+            valid_till=datetime.utcfromtimestamp(e["args"]["valid_till"]).isoformat(" "),
+            amount=e["args"]["deposit_amount"],
+            deposit_contract=e["args"]["deposit_contract"],
+            withdrawn=not web3.eth.getCode(e["args"]["deposit_contract"]),
+        )
+        for e in events
+    ]
+
+
+def find_withdrawable_deposit(
+    web3: Web3,
+    service_address: Address,
+    service_registry_contract: Contract,
+    start_block: BlockNumber,
+) -> Address:
+    # Get formatter for token amounts
+    deposit_token_address = service_registry_contract.functions.token().call()
+    deposit_token_contract = web3.eth.contract(
+        address=deposit_token_address,
+        abi=CONTRACT_MANAGER.get_contract_abi(CONTRACT_CUSTOM_TOKEN),
+    )
+    fmt_amount = get_token_formatter(deposit_token_contract)
+
+    # Find deposits
+    deposits = find_deposits(web3, service_address, service_registry_contract, start_block)
+    deposits = [d for d in deposits if not d["withdrawn"]]
+    if not deposits:
+        click.echo("No deposits found!", err=True)
+        sys.exit(1)
```

Comment:
Not finding withdrawable deposits does not necessarily constitute an error (`exit > 0`) to me. Does it to you?

Reply:
If withdraw is called, I would expect a withdraw to happen. It feels similar to an `rm` with a filename that does not exist to me. I do see your point, but prefer the current return code.